sdk/core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java

```diff
@@ -207,11 +207,68 @@ private void performLogging(LogLevel logLevel, LogLevel environmentLogLevel, boo
              * Environment is logging at a level higher than verbose, strip out the throwable as it would log its
              * stack trace which is only expected when logging at a verbose level.
              */
-            if (environmentLogLevel.getLogLevel() > LogLevel.VERBOSE.getLogLevel()) {
+            if (rootLogLevel.getLogLevel() > LogLevel.VERBOSE.getLogLevel()) {
                 args = removeThrowable(args);
             }
         }
+        boolean isFromEnv = isLogFromEnv();
+        printInPlace(isFromEnv, throwableMessage, logLevel, format, args);
+
+    }
+
+    private void printInPlace(boolean isFromEnv, String throwableMessage, LogLevel logLevel, String format,
+        Object... args) {
+        if (isFromEnv) {
+            printFromDefaultLogger(throwableMessage, logLevel, format, args);
+        } else {
+            printFromLogger(throwableMessage, logLevel, format, args);
+        }
+    }
+
+    private void printFromDefaultLogger(String throwableMessage, LogLevel logLevel, String format, Object... args) {
+        final DateFormat dateFormat = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss");
+        Date date = new Date();
+        String dateTime = dateFormat.format(date);
+        String classPath = className.getClass().getCanonicalName();
+
+        String message = String.format(format, args);
 
+        switch (logLevel) {
+            case VERBOSE:
+                String verboseMessage = String.format(LOG_MESSAGE_FORMAT, LOG_LEVEL_VERBOSE, dateTime, classPath,
+                    message);
+                System.out.println(verboseMessage);
+                break;
+            case INFORMATIONAL:
+                String infoMessage = String.format(LOG_MESSAGE_FORMAT, LOG_LEVEL_INFO, dateTime, classPath, message);
+                System.out.println(infoMessage);
+                break;
+            case WARNING:
+                if (!CoreUtils.isNullOrEmpty(throwableMessage)) {
+                    message += System.lineSeparator() + throwableMessage;
+                }
+
+                String warnMessage = String.format(LOG_MESSAGE_FORMAT, LOG_LEVEL_WARN, dateTime, classPath, message);
+                System.out.println(warnMessage);
+                break;
+            case ERROR:
+                if (!CoreUtils.isNullOrEmpty(throwableMessage)) {
+                    message += System.lineSeparator() + throwableMessage;
+                }
+                String errorMessage = String.format(LOG_MESSAGE_FORMAT, LOG_LEVEL_ERROR, dateTime, classPath, message);
+                System.out.println(errorMessage);
+                break;
+            default:
+                // Don't do anything, this state shouldn't be possible.
+                break;
+        }
```

Comment:
There is a lot of repetition in here - it looks like you could build the string and do the println outside of the switch statement.

Reply:
Move out some common stuff to printInPlace method.
Exact system out in end of the printFromDefault